karpenter-operator/controllers/karpenter/karpenter_controller.go

```diff
@@ -208,7 +214,70 @@ func (r *Reconciler) reconcileCRDs(ctx context.Context, onlyCreate bool) error {
 	return nil
 }
 
-func (r *Reconciler) reconcileEC2NodeClassOwnedFields(ctx context.Context, userDataSecret *corev1.Secret) error {
+func (r *Reconciler) reconcileVAP(ctx context.Context) error {
+	vap := &admissionv1.ValidatingAdmissionPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "karpenter.hypershift.io",
+		},
+	}
+
+	if _, err := r.CreateOrUpdate(ctx, r.GuestClient, vap, func() error {
+		vap.Spec.MatchConstraints = &admissionv1.MatchResources{
+			ResourceRules: []admissionv1.NamedRuleWithOperations{
+				{
+					RuleWithOperations: admissionv1.RuleWithOperations{
+						Operations: []admissionv1.OperationType{
+							admissionv1.Update,
+						},
+						Rule: admissionv1.Rule{
+							APIGroups:   []string{"karpenter.k8s.aws"},
```

Comment:
nit: we are referencing karpenter.k8s.aws multiple times in this field now, might be good to move to a constant